corehq/apps/users/tests/test_user_data.py

```diff
@@ -289,3 +312,121 @@ def test_remove_unrecognized_empty_field(self):
+            'commcare_location_id': self.loc2.location_id,
+            'commcare_location_ids': f"{self.loc1.location_id} {self.loc2.location_id}",
+            'commcare_primary_case_sharing_id': self.loc2.location_id,
+        })
+
+    def test_append_location_and_user_data_updated(self):
+        # Set primary location
+        self.user.set_location(self.loc1)
+        user_data = self.user.get_user_data(self.domain)
+        self.assertEqual(user_data.to_dict(), {
+            'commcare_project': self.domain,
+            'commcare_profile': '',
+            'commcare_location_id': self.loc1.location_id,
+            'commcare_location_ids': self.loc1.location_id,
+            'commcare_primary_case_sharing_id': self.loc1.location_id,
+        })
+
+        # Append a location but not change primary location
+        self.user.add_to_assigned_locations(self.loc2)
+
+        self.assertEqual(user_data.to_dict(), {
+            'commcare_project': self.domain,
+            'commcare_profile': '',
+            'commcare_location_id': self.loc1.location_id,
+            'commcare_location_ids': f"{self.loc1.location_id} {self.loc2.location_id}",
+            'commcare_primary_case_sharing_id': self.loc1.location_id,
+        })
+
+    def test_remove_primary_location_and_user_data_updated(self):
+        # Set primary location
+        self.user.set_location(self.loc1)
+        user_data = self.user.get_user_data(self.domain)
+        self.assertEqual(user_data.to_dict(), {
+            'commcare_project': self.domain,
+            'commcare_profile': '',
+            'commcare_location_id': self.loc1.location_id,
+            'commcare_location_ids': self.loc1.location_id,
+            'commcare_primary_case_sharing_id': self.loc1.location_id,
+        })
+
+        self.user.unset_location()
+
+        self.assertEqual(user_data.to_dict(), {
+            'commcare_project': self.domain,
+            'commcare_profile': '',
+        })
+
+    def test_remove_assigned_location_and_user_data_updated(self):
+        self.user.reset_locations(self.loc_ids)
+        user_data = self.user.get_user_data(self.domain)
+        self.assertEqual(user_data.to_dict(), {
+            'commcare_project': self.domain,
+            'commcare_profile': '',
+            'commcare_location_id': self.loc1.location_id,
+            'commcare_location_ids': f"{self.loc1.location_id} {self.loc2.location_id}",
+            'commcare_primary_case_sharing_id': self.loc1.location_id,
+        })
+
+        self.user.unset_location_by_id(self.loc2.location_id)
+
+        self.assertEqual(user_data.to_dict(), {
+            'commcare_project': self.domain,
+            'commcare_profile': '',
+            'commcare_location_id': self.loc1.location_id,
+            'commcare_location_ids': f"{self.loc1.location_id}",
+            'commcare_primary_case_sharing_id': self.loc1.location_id,
+        })
+
+    def test_reset_locations_and_user_data_updated(self):
```

Comment:
I see that line of reasoning. Since it's testing the effect of the action on the user data, what would you think about phrasing it as `updates_user_data` instead? To me, that reads much more clearly, like "performing x action has y result", where the result is that it updates `user_data`.

Reply:
`udpates_user_data` sounds more natural! Thank you! Updated!